flink-clients/src/main/java/org/apache/flink/client/program/rest/RestClusterClient.java

```diff
@@ -608,6 +616,27 @@ private CompletableFuture<String> triggerSavepoint(
                         });
     }
 
+    private CompletableFuture<String> triggerDetachSavepoint(
+            final JobID jobId,
+            final @Nullable String savepointDirectory,
+            final boolean cancelJob,
+            final SavepointFormatType formatType) {
+        final SavepointTriggerHeaders savepointTriggerHeaders =
```

Comment:
The previous logic is similar with savepoint, right ?
Could we unify them into a method ?